src/OrchardCore/OrchardCore.ContentManagement.Display/ContentDisplay/ContentFieldDisplayDriver.cs

```diff
@@ -10,6 +11,76 @@ namespace OrchardCore.ContentManagement.Display.ContentDisplay
 {
     public abstract class ContentFieldDisplayDriver<TField> : DisplayDriverBase, IContentFieldDisplayDriver where TField : ContentField, new()
     {
+        private ContentTypePartDefinition _typePartDefinition;
+        private ContentPartFieldDefinition _partFieldDefinition;
+
+        public override ShapeResult Shape(string shapeType, Func<IBuildShapeContext, Task<IShape>> shapeBuilder, Func<IShape, Task> initializeAsync)
+        {
+            // e.g., BodyPart.Summary, BodyPart-BlogPost, BagPart-LandingPage-Services
+            // context.Shape is the ContentItem shape, we need to alter the part shape
+
+            var result = base.Shape(shapeType, shapeBuilder, initializeAsync).Prefix(Prefix);
+
+            // This should only be set in Display methods
+            if (_typePartDefinition != null && _partFieldDefinition != null)
+            {
+                var partType = _typePartDefinition.PartDefinition.Name;
+                var partName = _typePartDefinition.Name;
+                var fieldType = _partFieldDefinition.FieldDefinition.Name;
+                var fieldName = _partFieldDefinition.Name;
+                var contentType = _typePartDefinition.ContentTypeDefinition.Name;
+
+                if (partType == shapeType)
+                {
+                    // BodyPart-Description, Services-Description
+                    result.Differentiator($"{partName}-{fieldName}");
+                }
+                else
+                {
+                    // BodyPart-Description-TextField, Services-Description-TextField
+                    result.Differentiator($"{partName}-{fieldName}-{shapeType}");
+                }
```

Comment:
Same remark about the `-` character in liquid syntax, see my above comment for the related code for parts where i suggest to directly use `_`. And see my other suggestion to check if `shapeType` starts with `fieldName` ...